crates/sui-core/src/authority/test_authority_builder.rs

```diff
@@ -317,19 +312,43 @@ impl<'a> TestAuthorityBuilder<'a> {
             SupportedProtocolVersions::SYSTEM_DEFAULT,
             authority_store,
             cache_traits,
-            epoch_store,
+            epoch_store.clone(),
             committee_store,
             index_store,
             rest_index,
             checkpoint_store,
             &registry,
             genesis.objects(),
             &DBCheckpointConfig::default(),
-            config,
+            config.clone(),
             usize::MAX,
             ArchiveReaderBalancer::default(),
         )
         .await;
+
+        // Set up randomness with no-op consensus (DKG will not complete).
+        if epoch_store.randomness_state_enabled() {
+            let consensus_client = Box::new(MockConsensusClient::new(
+                Arc::downgrade(&state),
+                ConsensusMode::Noop,
+            ));
+            let randomness_manager = RandomnessManager::try_new(
+                Arc::downgrade(&epoch_store),
+                consensus_client,
+                randomness::Handle::new_stub(),
+                config.protocol_key_pair(),
+            )
+            .await;
+            if let Some(randomness_manager) = randomness_manager {
```

Comment:
Is there a reason this may be None in tests?

Reply:
Yes, if it fails to construct for any reason. for example maybe the fake committee does not have the right properties to initiate DKG.

Reply:
And what is that property? Does that happen by accident or is there legit test cases where we want a a Committee without the right properties to initiate DKG?

Reply:
Fastcrypto code does various sanity checks on properties such as the validity thresholds that may not pass in a fake test env with dummy settings. There are def test cases where this failed bc I had to disable randomness for those test cases to keep working, since it then invoked consensus code that failed without a randomness mgr. There may be others where the test still works even if this can't be constructed because it doesn't invoke consensus code, but I didn't go and catalog every single one of the tests that uses TestAuthorityBuilder